nimbus/src/lib.rs

```diff
@@ -170,20 +201,29 @@ impl NimbusClient {
 
     pub fn set_experiments_locally(&self, experiments_json: String) -> Result<()> {
         let new_experiments = parse_experiments(&experiments_json)?;
-        write_pending_experiments(self.db()?, new_experiments)?;
+        let db = self.db()?.lock().unwrap();
+        write_pending_experiments(&db, new_experiments)?;
         Ok(())
     }
 
     pub fn nimbus_id(&self) -> Result<Uuid> {
-        let db = self.db()?;
+        let db = self.db()?.lock().unwrap();
         let mut writer = db.write()?;
+        let uuid = self.read_or_create_nimbus_id(&db, &mut writer)?;
+        // We don't know whether we needed to generate and save the uuid, so
+        // we commit just in case - this is hopefully close to a noop in that
+        // case!
```

Comment:
You could return a `(nimbus_id, was_created)` tuple if you really wanted to know...but I agree it should be cheap in practice.

Reply:
yeah, doesn't seem worth the complexity